### PR TITLE
Enhanced Open Source project infrastructure.

### PR DESCRIPTION
**Pull request by a11n**
Refs #176

### CODE REVIEW

**Comment by alexeyOnGitHub** [comment]
I will take a look at this

**Comment by alexeyOnGitHub** [comment]
how about splitting these changes to separate pull requests? they are mostly  independent...

File: build.gradle [hunk 1]
```diff
@@ -114,3 +114,31 @@ artifacts {
...
+  repoUri = 'https://github.com/taskadapter/redmine-java-api'
+  deleteExistingFiles true
+  pages {
+    from javadoc.outputs.files
+  }
+}
+
+task generateJavadocAndPushToGhPages{
+  dependsOn javadoc
+  dependsOn publishGhPages
```
Comment (alexeyOnGitHub) [inline build.gradle]: this brought a window on my desktop asking for login/password without specifying what they are for. I assume they are for Github, but it looks weird... 
this command has to accept API keys...

**Comment by a11n** [comment]
Sorry, my fault to put too many commits in one PR.

I see you already adopted some of my proposals :-) So, the only new contribution would be the Javadoc generation.

I will close this PR, rework the solution and do another PR.

**Pull request closed without merging**